pkg/tagger/collectors/docker_extract.go

```diff
@@ -8,66 +8,99 @@
 func (c *DockerCollector) extractFromInspect(co types.ContainerJSON) ([]string, []string, error) {
-	var low, high []string
+	ex := newExtractor(c)
 
-	docker_image, err := c.dockerUtil.ResolveImageName(co.Image)
-	if err != nil {
-		log.Debugf("error resolving image %s: %s", co.Image, err)
+	ex.extractImage(co)
+	ex.extractLabels(co.Config.Labels)
+	ex.extractEnvironmentVariables(co.Config.Env)
 
-	} else {
-		image_name, _, image_tag, err := docker.SplitImageName(docker_image)
+	ex.tags.AddHigh("container_name", strings.TrimPrefix(co.Name, "/"))
+	ex.tags.AddHigh("container_id", co.ID)
 
-		low = append(low, fmt.Sprintf("docker_image:%s", docker_image))
+	low, high := ex.tags.Compute()
+	return low, high, nil
+}
 
-		if err != nil {
-			log.Debugf("error spliting %s: %s", docker_image, err)
-		} else {
-			low = append(low, fmt.Sprintf("image_name:%s", image_name))
-			low = append(low, fmt.Sprintf("image_tag:%s", image_tag))
-		}
+func (e *extract) extractImage(co types.ContainerJSON) {
+	dockerImage, err := e.dockerCollector.dockerUtil.ResolveImageName(co.Image)
+	if err != nil {
+		log.Debugf("error resolving image %s: %s", co.Image, err)
+		return
+	}
+	imageName, _, imageTag, err := docker.SplitImageName(dockerImage)
+	e.tags.AddLow("docker_image", dockerImage)
+	if err != nil {
+		log.Debugf("error splitting %s: %s", dockerImage, err)
+		return
 	}
+	e.tags.AddLow("image_name", imageName)
+	e.tags.AddLow("image_tag", imageTag)
+}
 
-	if len(c.labelsAsTags) > 0 {
-		for label_name, label_value := range co.Config.Labels {
-			if tag_name, found := c.labelsAsTags[strings.ToLower(label_name)]; found {
-				if tag_name[0] == '+' {
-					high = append(high, fmt.Sprintf("%s:%s", tag_name[1:], label_value))
-				} else {
-					low = append(low, fmt.Sprintf("%s:%s", tag_name, label_value))
-				}
+func (e *extract) extractLabels(labels map[string]string) {
+	for labelName, labelValue := range labels {
+		if tagName, found := e.dockerCollector.labelsAsTags[strings.ToLower(labelName)]; found {
+			if tagName[0] == '+' {
+				e.tags.AddHigh(tagName[1:], labelValue)
+				continue
 			}
+			e.tags.AddLow(tagName, labelValue)
 		}
 	}
+}
 
-	if len(c.envAsTags) > 0 {
-		for _, envvar := range co.Config.Env {
-			parts := strings.SplitN(envvar, "=", 2)
-			if len(parts) != 2 {
-				continue
-			}
-			if tag_name, found := c.envAsTags[strings.ToLower(parts[0])]; found {
-				if tag_name[0] == '+' {
-					high = append(high, fmt.Sprintf("%s:%s", tag_name[1:], parts[1]))
-				} else {
-					low = append(low, fmt.Sprintf("%s:%s", tag_name, parts[1]))
+// extractEnvironmentVariables contain hard-coded environment variables from:
+// - Mesos Marathon
+func (e *extract) extractEnvironmentVariables(envVariables []string) {
+	var envSplit []string
+
+	for _, envEntry := range envVariables {
+		envSplit = strings.SplitN(envEntry, "=", 2)
+		if len(envSplit) != 2 {
+			continue
+		}
+		switch envSplit[0] {
+		// Mesos Marathon
```

Comment:
There's also Chronos tasks, so you could put `Mesos/DCOS tags (mesos, marathon, chronos)`